main.go

```diff
@@ -61,62 +53,169 @@ func List() {
-	// how to encode with user input???
-	f, err := os.Open("phonebook.json")
+func Add() {
+	// how to encode with user input??? Tutorial by me (version 1)
+	var entries []Entry // make variable where will be all data
+
+	f, err := os.Open("phonebook.json") // open and encode file with already been information and put it in variable you made in previous step
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
+	if err := json.NewDecoder(bufio.NewReader(f)).Decode(&entries); err != nil {
+		log.Fatal(err)
+	}
+
+	var newentries Entry // make new variable with type of your struct, but be careful - it need to be NOT SLICE OF STRUCT, JUST STRUCT, CAUSE WE MAKING ONLY ONE NEW CONTACT, NOT 5 IN ONE TIME
+
+	scanner := bufio.NewScanner(os.Stdin) // Scan information you need
+	fmt.Println("Enter last name:")       // better not use bufio reader, it add \n\r to text and you ll need 2 variables for string and error
+	scanner.Scan()
+	scanner.Scan()
+	newentries.LastName = scanner.Text()
+	fmt.Println("Enter first name:")
+	scanner.Scan()
+	newentries.FirstName = scanner.Text()
+	fmt.Println("Enter phone number:")
+	scanner.Scan()
+	newentries.PhoneNumber = scanner.Text()
+	newentries.ID = GiveID(entries)
+
+	entries = append(entries, newentries) // add scanned data to decoded earlier data
+
+	f, err = os.Create("phonebook.json") // and encode it back
 	if err != nil {
 		log.Fatal(err)
 	}
 	defer f.Close()
 	w := bufio.NewWriter(f)
 	defer w.Flush()
-	//reader := bufio.NewReader(os.Stdin)
 	if err := json.NewEncoder(w).Encode(entries); err != nil {
 		log.Fatal(err)
 	}
 
 }
 
-///func Remove(entries []Entry, s int) []Entry {
-// return append(s[:s], slice[s+1:]...)
-//}
+func Remove() {
+	var entries []Entry                 //var
+	f, err := os.Open("phonebook.json") // opening and decoding
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
+	if err := json.NewDecoder(bufio.NewReader(f)).Decode(&entries); err != nil {
+		log.Fatal(err)
+	}
+	var noid, nosure uint // actions, work with data
+	fmt.Println("Please, enter ID of contact you want to delete:")
+	fmt.Scan(&noid)
+	for i, j := range entries {
+		if j.ID == noid {
+			fmt.Println(`Are you sure? 
+*the company "Ltd Littleunidragon" is not responsible for "accidentally" permanently deleted contacts
+1) Yes
+2) No `)
+			fmt.Scan(&nosure)
+			if nosure == 1 {
+				entries = append(entries[:i], entries[i+1:]...)
+				fmt.Println("Contact with ID ", noid, " was removed")
+				break
+			} else {
+				break
+			}
+		}
+	}
+	f, err = os.Create("phonebook.json") // and encode it back
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
+	w := bufio.NewWriter(f)
+	defer w.Flush()
+	if err := json.NewEncoder(w).Encode(entries); err != nil {
+		log.Fatal(err)
+	}
+}
 
 func Exit() {
 	os.Exit(0)
 }
 
-func Feedback() {
-	var stars int
-	fmt.Println("How many stars would you give us on Play Market?")
-	fmt.Scan(&stars)
-	if stars < 0 {
+type Feedbackstruct struct {
+	Stars int
+	Text  string
```

Comment:
As a challenge - make it so that `Feedbackstruct.Text` is not stored if empty (-; (we've discussed it today)